packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+            <HistoryIcon isHistory={isHistory} />
+        </div>
+    );
+};
+
+const HistoryIcon = ({isHistory}: {isHistory: boolean}) => {
+    return (
+        isHistory && (
+            <span className="icon mod-lg limit-history-button">
+                <Svg svgName={VaporSVG.svg.menuAnalytics.name} className="fill-medium-blue" />
+            </span>
+        )
+    );
+};
+
+const ContentDivision = ({
+    id,
+    usageValue,
+    limitValue,
+    isEditable,
+    limitTitle,
+    onChangeLimit,
+}: {
+    id: string;
+    usageValue: number;
+    limitValue: number;
+    isEditable: boolean;
+    limitTitle: string;
+    onChangeLimit: (limit: string) => void;
+}) => {
+    return (
+        <div className="limit-box-numbers pt1 flex">
+            <UsageDivision usageValue={usageValue} />
+            <LimitDivision
+                id={id}
+                onChangeLimit={onChangeLimit}
+                usageValue={usageValue}
+                limitValue={limitValue}
+                isEditable={isEditable}
+                limitTitle={limitTitle}
+            />
+        </div>
+    );
+};
+
+const UsageDivision = ({usageValue}: {usageValue: number}) => {
+    const isUsageRequired: boolean = !!usageValue;
+    return (
+        isUsageRequired && (
+            <div className="limit-box-usage">
+                <label className="form-control-label">Usage</label>
+                <span className="limit-box-usage-value">{usageValue}</span>
+            </div>
+        )
+    );
+};
+
+const LimitDivision = ({
+    id,
+    limitValue,
+    usageValue,
+    isEditable,
+    limitTitle,
+    onChangeLimit,
+}: {
+    id: string;
+    limitValue: number;
+    usageValue: number;
+    isEditable: boolean;
+    limitTitle: string;
+    onChangeLimit: (limit: string) => void;
+}) => {
+    const limitValueString: string = limitValue.toString();
+    const minLimitValue: number = usageValue || 0;
+    if (isEditable) {
+        return (
+            <InputConnected
+                id={id}
+                type="number"
+                labelTitle={limitTitle}
+                defaultValue={limitValueString}
+                min={minLimitValue}
+                classes="limit-box-limit form-group input-field validate"
+                onChange={(limit) => onChangeLimit(limit)}
+            ></InputConnected>
+        );
+    } else {
+        return (
+            <div className="limit-box-limit">
+                <label className="form-control-label">{limitTitle}</label>
+                <span className="limit-box-limit-value">{limitValue}</span>
+            </div>
+        );
+    }
+};
+
+const ProgressBar = ({
+    usageValue,
+    isLimitTheGoal,
+    limitValue,
+}: {
+    usageValue: number;
+    isLimitTheGoal: boolean;
+    limitValue: number;
+}) => {
+    const isUsageRequired: boolean = !!usageValue;
+    const progressClass: string = getProgressBarClass({usageValue, limitValue});
```

Comment:
I think you could avoid structuring/destructuring you object even if it is fun ;) since you have only two args in the params.

`getProgressBarClass(usageValue, limitValue)`

`const getProgressBarClass = (usageValue: number, limitValue:number): string => {}`